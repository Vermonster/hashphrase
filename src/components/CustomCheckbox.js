import React from 'react';
import { View, StyleSheet, TouchableOpacity } from 'react-native';
import { IconButton } from 'react-native-paper';
import { colors } from '../styles/base';

const styles = StyleSheet.create({
  box: {
    height: 30,
    width: 30,
    borderRadius: 3,
    borderWidth: 3,
    marginRight: 20,
    borderColor: colors.secondary,
    display: 'flex',
    justifyContent: 'space-around',
    alignItems: 'center',
  },
});

const CustomCheckbox = (props) => {
  const { status, handlePress } = props;
  const checkBox = (
    <TouchableOpacity onPress={handlePress}>
      <View style={styles.box}>
        {status && <IconButton color={colors.secondary} icon="done" />}
      </View>
    </TouchableOpacity>
  );
  return checkBox;
};

export default CustomCheckbox;
