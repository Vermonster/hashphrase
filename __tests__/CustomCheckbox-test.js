import React from 'react';
import { shallow } from 'enzyme';
import CustomCheckbox from '../src/components/CustomCheckbox';

describe('<CustomCheckbox />', () => {
  let wrapper;
  beforeEach(() => {
    const mockFn = jest.fn();
    wrapper = shallow(
      <CustomCheckbox
        testID="clear-clipboard"
        status={false}
        closeModal={mockFn}
      />,
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('only shows checkmark when status is true', () => {
    expect(wrapper.find('Icon')).toHaveLength(0);
    wrapper.setProps({ status: true });
    expect(wrapper.find('Icon')).toHaveLength(1);
  });
});
