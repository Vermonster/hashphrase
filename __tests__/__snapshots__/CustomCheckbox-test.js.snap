// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomCheckbox /> should match the snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
>
  <ForwardRef(View)
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#212A59",
        "borderRadius": 3,
        "borderWidth": 3,
        "display": "flex",
        "height": 30,
        "justifyContent": "space-around",
        "marginRight": 20,
        "width": 30,
      }
    }
  />
</TouchableOpacity>
`;
